### ISSUE TITLE
RuntimeError: stack expects each tensor to be equal size, but got [34] at entry 0 and [30] at entry 1

### ISSUE THREAD

**Issue by mingzailao** [nicolas-chaulet/torch-points3d#437]
When I Run: 
```bash
python train.py task=segmentation model_type=pointnet2 model_name=pointnet2_largemsg dataset=s3disfused-fixed training=s3dis_benchmark/pointnet2
```
```bash
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/root/.cache/pypoetry/virtualenvs/torch-points3d-v1fb3WC--py3.6/lib/python3.6/site-packages/torch/utils/data/_utils/fetch.py", line 47, in fetch
    return self.collate_fn(data)
  File "/root/torch-points3d/torch_points3d/datasets/batch.py", line 48, in from_data_list
    batch[key] = torch.stack(batch[key])
RuntimeError: stack expects each tensor to be equal size, but got [34] at entry 0 and [30] at entry 1
```

Maybe I Use different data?
During data processing(patch)

```bash
patching file Area_1/WC_1/WC_1.txt
Hunk #1 FAILED at 429610.
1 out of 1 hunk FAILED -- saving rejects to file uN
patching file Area_6/hallway_2/hallway_2.txt
Hunk #1 FAILED at 842730.
1 out of 1 hunk FAILED -- saving rejects to file uN
```

I Just use data and create a soft link in ./data/s3disfused/
```bash
-rw-r--r--  1 root root  5149528240 Sep 22 09:05 Stanford3dDataset_v1.2.zip
```
```bash
lrwxrwxrwx 1 root root   35 Sep 23 01:48 Stanford3dDataset_v1.2_Version.zip -> /data/3D/Stanford3dDataset_v1.2.zip
```
it has different size in https://github.com/nicolas-chaulet/torch-points3d/blob/master/torch_points3d/datasets/segmentation/s3dis.md

can you help me analysis the reason?

**Comment by nicolas-chaulet** [comment]
That's quite weird. Have you changed anything in the config files?

**Comment by mingzailao** [comment]
No
There is my env:

Docker :  Nvidia: Cuda10.2 + cudnn7
Use Poetry install all package

In conf file, I just change entity: from "nicolas" to my own.
nothing else
```python
In [1]: import torch; print(torch.version.cuda)
10.2
```

can i do anything for you to debug it ？

**Comment by mingzailao** [comment]
Here is the fully output:
```bash
DEVICE : cuda
wandb: Tracking run with wandb version 0.8.36
wandb: Wandb version 0.10.2 is available!  To upgrade, please run:
wandb:  $ pip install wandb --upgrade
wandb: Run data is saved locally in wandb/run-20200923_123254-2clzshtb
wandb: Syncing run pointnet2
wandb: ⭐️ View project at https://app.wandb.ai/mingzailao/s3dis-benchmark
wandb: 🚀 View run at https://app.wandb.ai/mingzailao/s3dis-benchmark/runs/2clzshtb
wandb: Run `wandb off` to turn off syncing.

100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:18<00:00,  3.79s/it]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:21<00:00, 21.77s/it]
The model will not be able to be used from pretrained weights without the corresponding dataset.
Setting batchnorm momentum at 0.98
PointNet2_D(
  (model): UnetSkipConnectionBlock(
    (down): PointNetMSGDown(
      (mlps): ModuleList(
        (0): MLP2D(
          (0): Conv2D(
            (0): Conv2d(7, 16, kernel_size=(1, 1), stride=(1, 1), bias=False)
            (1): BatchNorm2d(16, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
            (2): LeakyReLU(negative_slope=0.01)
          )
          (1): Conv2D(
            (0): Conv2d(16, 16, kernel_size=(1, 1), stride=(1, 1), bias=False)
            (1): BatchNorm2d(16, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
            (2): LeakyReLU(negative_slope=0.01)
          )
          (2): Conv2D(
            (0): Conv2d(16, 32, kernel_size=(1, 1), stride=(1, 1), bias=False)
            (1): BatchNorm2d(32, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
            (2): LeakyReLU(negative_slope=0.01)
          )
        )
        (1): MLP2D(
          (0): Conv2D(
            (0): Conv2d(7, 32, kernel_size=(1, 1), stride=(1, 1), bias=False)
            (1): BatchNorm2d(32, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
            (2): LeakyReLU(negative_slope=0.01)
          )
          (1): Conv2D(
            (0): Conv2d(32, 32, kernel_size=(1, 1), stride=(1, 1), bias=False)
            (1): BatchNorm2d(32, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
            (2): LeakyReLU(negative_slope=0.01)
          )
          (2): Conv2D(
            (0): Conv2d(32, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
            (1): BatchNorm2d(64, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
            (2): LeakyReLU(negative_slope=0.01)
          )
        )
      )
    )
    (submodule): UnetSkipConnectionBlock(
      (down): PointNetMSGDown(
        (mlps): ModuleList(
          (0): MLP2D(
            (0): Conv2D(
              (0): Conv2d(99, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
              (1): BatchNorm2d(64, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
              (2): LeakyReLU(negative_slope=0.01)
            )
            (1): Conv2D(
              (0): Conv2d(64, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
              (1): BatchNorm2d(64, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
              (2): LeakyReLU(negative_slope=0.01)
            )
            (2): Conv2D(
              (0): Conv2d(64, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
              (1): BatchNorm2d(128, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
              (2): LeakyReLU(negative_slope=0.01)
            )
          )
          (1): MLP2D(
            (0): Conv2D(
              (0): Conv2d(99, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
              (1): BatchNorm2d(64, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
              (2): LeakyReLU(negative_slope=0.01)
            )
            (1): Conv2D(
              (0): Conv2d(64, 96, kernel_size=(1, 1), stride=(1, 1), bias=False)
              (1): BatchNorm2d(96, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
              (2): LeakyReLU(negative_slope=0.01)
            )
            (2): Conv2D(
              (0): Conv2d(96, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
              (1): BatchNorm2d(128, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
              (2): LeakyReLU(negative_slope=0.01)
            )
          )
        )
      )
      (submodule): UnetSkipConnectionBlock(
        (down): PointNetMSGDown(
          (mlps): ModuleList(
            (0): MLP2D(
              (0): Conv2D(
                (0): Conv2d(259, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
                (1): BatchNorm2d(128, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                (2): LeakyReLU(negative_slope=0.01)
              )
              (1): Conv2D(
                (0): Conv2d(128, 196, kernel_size=(1, 1), stride=(1, 1), bias=False)
                (1): BatchNorm2d(196, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                (2): LeakyReLU(negative_slope=0.01)
              )
              (2): Conv2D(
                (0): Conv2d(196, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
                (1): BatchNorm2d(256, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                (2): LeakyReLU(negative_slope=0.01)
              )
            )
            (1): MLP2D(
              (0): Conv2D(
                (0): Conv2d(259, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
                (1): BatchNorm2d(128, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                (2): LeakyReLU(negative_slope=0.01)
              )
              (1): Conv2D(
                (0): Conv2d(128, 196, kernel_size=(1, 1), stride=(1, 1), bias=False)
                (1): BatchNorm2d(196, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                (2): LeakyReLU(negative_slope=0.01)
              )
              (2): Conv2D(
                (0): Conv2d(196, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
                (1): BatchNorm2d(256, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                (2): LeakyReLU(negative_slope=0.01)
              )
            )
          )
        )
        (submodule): UnetSkipConnectionBlock(
          (down): PointNetMSGDown(
            (mlps): ModuleList(
              (0): MLP2D(
                (0): Conv2D(
                  (0): Conv2d(515, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
                  (1): BatchNorm2d(256, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                  (2): LeakyReLU(negative_slope=0.01)
                )
                (1): Conv2D(
                  (0): Conv2d(256, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
                  (1): BatchNorm2d(256, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                  (2): LeakyReLU(negative_slope=0.01)
                )
                (2): Conv2D(
                  (0): Conv2d(256, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
                  (1): BatchNorm2d(512, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                  (2): LeakyReLU(negative_slope=0.01)
                )
              )
              (1): MLP2D(
                (0): Conv2D(
                  (0): Conv2d(515, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
                  (1): BatchNorm2d(256, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                  (2): LeakyReLU(negative_slope=0.01)
                )
                (1): Conv2D(
                  (0): Conv2d(256, 384, kernel_size=(1, 1), stride=(1, 1), bias=False)
                  (1): BatchNorm2d(384, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                  (2): LeakyReLU(negative_slope=0.01)
                )
                (2): Conv2D(
                  (0): Conv2d(384, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
                  (1): BatchNorm2d(512, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
                  (2): LeakyReLU(negative_slope=0.01)
                )
              )
            )
          )
          (submodule): Identity()
          (up): DenseFPModule: 1050624 (MLP2D(
            (0): Conv2D(
              (0): Conv2d(1536, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
              (1): BatchNorm2d(512, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
              (2): LeakyReLU(negative_slope=0.01)
            )
            (1): Conv2D(
              (0): Conv2d(512, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
              (1): BatchNorm2d(512, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
              (2): LeakyReLU(negative_slope=0.01)
            )
          ))
        )
        (up): DenseFPModule: 657408 (MLP2D(
          (0): Conv2D(
            (0): Conv2d(768, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
            (1): BatchNorm2d(512, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
            (2): LeakyReLU(negative_slope=0.01)
          )
          (1): Conv2D(
            (0): Conv2d(512, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
            (1): BatchNorm2d(512, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
            (2): LeakyReLU(negative_slope=0.01)
          )
        ))
      )
      (up): DenseFPModule: 222208 (MLP2D(
        (0): Conv2D(
          (0): Conv2d(608, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (1): BatchNorm2d(256, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
          (2): LeakyReLU(negative_slope=0.01)
        )
        (1): Conv2D(
          (0): Conv2d(256, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
          (1): BatchNorm2d(256, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
          (2): LeakyReLU(negative_slope=0.01)
        )
      ))
    )
    (up): DenseFPModule: 50176 (MLP2D(
      (0): Conv2D(
        (0): Conv2d(260, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (1): BatchNorm2d(128, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
        (2): LeakyReLU(negative_slope=0.01)
      )
      (1): Conv2D(
        (0): Conv2d(128, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (1): BatchNorm2d(128, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
        (2): LeakyReLU(negative_slope=0.01)
      )
    ))
  )
  (FC_layer): Seq(
    (0): Conv1D(
      (0): Conv1d(128, 128, kernel_size=(1,), stride=(1,), bias=False)
      (1): BatchNorm1d(128, eps=1e-05, momentum=0.98, affine=True, track_running_stats=True)
      (2): LeakyReLU(negative_slope=0.01)
    )
    (1): Dropout(p=0.5, inplace=False)
    (2): Conv1D(
      (0): Conv1d(128, 13, kernel_size=(1,), stride=(1,))
    )
  )
)
Optimizer: SGD (
Parameter Group 0
    dampening: 0
    initial_lr: 0.01
    lr: 0.01
    momentum: 0.98
    nesterov: False
    weight_decay: 0.001
)
Learning Rate Scheduler: ExponentialLR({'gamma': 0.9885}, update_scheduler_on=on_epoch)
BatchNorm Scheduler: BNMomentumScheduler(base_momentum: 0.98, update_scheduler_on=on_epoch)
Accumulated gradients: None
Model size = 3026477
Dataset: S3DISFusedDataset
pre_transform = None
test_transform = Compose([
    FixedPoints(20000, replace=True),
    XYZFeature(axis=['z']),
    AddFeatsByKeys(rgb=True, pos_z=True),
    Center(),
    ScalePos(scale=0.5),
])
train_transform = Compose([
    FixedPoints(20000, replace=True),
    RandomNoise(sigma=0.001, clip=0.05),
    RandomRotate((-180, 180), axis=2),
    RandomScaleAnisotropic([0.8, 1.2]),
    Random symmetry of axes: x=True, y=False, z=False,
    DropFeature: proba = 0.2, feature = rgb,
    XYZFeature(axis=['z']),
    AddFeatsByKeys(rgb=True, pos_z=True),
    Center(),
    ScalePos(scale=0.5),
])
val_transform = Compose([
    FixedPoints(20000, replace=True),
    XYZFeature(axis=['z']),
    AddFeatsByKeys(rgb=True, pos_z=True),
    Center(),
    ScalePos(scale=0.5),
])
inference_transform = Compose([
    FixedPoints(20000, replace=True),
    XYZFeature(axis=['z']),
    AddFeatsByKeys(rgb=True, pos_z=True),
    Center(),
    ScalePos(scale=0.5),
])
pre_collate_transform = Compose([
    PointCloudFusion(),
    SaveOriginalPosId,
    GridSampling3D(grid_size=0.04, quantize_coords=False, mode=mean),
])
Size of train_dataset = 3000
Size of test_dataset = 1679
Size of val_dataset = 1166
Batch size = 8
Available stage selection datasets:  ['test', 'val']
The models will be selected using the metrics on following dataset:  val
EPOCH 1 / 300
  0%|                                                                                                                                                | 0/375 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "train.py", line 21, in <module>
    main()
  File "/root/.cache/pypoetry/virtualenvs/torch-points3d-v1fb3WC--py3.6/lib/python3.6/site-packages/hydra/main.py", line 24, in decorated_main
    strict=strict,
  File "/root/.cache/pypoetry/virtualenvs/torch-points3d-v1fb3WC--py3.6/lib/python3.6/site-packages/hydra/_internal/utils.py", line 174, in run_hydra
    overrides=args.overrides,
  File "/root/.cache/pypoetry/virtualenvs/torch-points3d-v1fb3WC--py3.6/lib/python3.6/site-packages/hydra/_internal/hydra.py", line 86, in run
    job_subdir_key=None,
  File "/root/.cache/pypoetry/virtualenvs/torch-points3d-v1fb3WC--py3.6/lib/python3.6/site-packages/hydra/plugins/common/utils.py", line 109, in run_job
    ret.return_value = task_function(task_cfg)
  File "train.py", line 13, in main
    trainer.train()
  File "/root/torch-points3d/torch_points3d/trainer.py", line 136, in train
    self._train_epoch(epoch)
  File "/root/torch-points3d/torch_points3d/trainer.py", line 186, in _train_epoch
    for i, data in enumerate(tq_train_loader):
  File "/root/.cache/py
wandb: Waiting for W&B process to finish, PID 10113
poetry/virtualenvs/torch-points3d-v1fb3WC--py3.6/lib/python3.6/site-packages/tqdm/std.py", line 1130, in __iter__
    for obj in iterable:
  File "/root/.cache/pypoetry/virtualenvs/torch-points3d-v1fb3WC--py3.6/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 363, in __next__
    data = self._next_data()
  File "/root/.cache/pypoetry/virtualenvs/torch-points3d-v1fb3WC--py3.6/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 403, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/root/.cache/pypoetry/virtualenvs/torch-points3d-v1fb3WC--py3.6/lib/python3.6/site-packages/torch/utils/data/_utils/fetch.py", line 47, in fetch
    return self.collate_fn(data)
  File "/root/torch-points3d/torch_points3d/datasets/batch.py", line 48, in from_data_list
    batch[key] = torch.stack(batch[key])
RuntimeError: stack expects each tensor to be equal size, but got [30] at entry 0 and [17] at entry 1
wandb: Program failed with code 1. Press ctrl-c to abort syncing.
wandb: Syncing files in wandb/run-20200923_123254-2clzshtb:
wandb:   change.patch
wandb:   hydra-config.yaml
wandb:   overrides.yaml
wandb: plus 7 W&B file(s) and 0 media file(s)
wandb:
wandb: Synced pointnet2: https://app.wandb.ai/mingzailao/s3dis-benchmark/runs/2clzshtb
```

**Comment by nicolas-chaulet** [comment]
Hi, I just launched the training on my end and all is good... It must be coming from the dataset that might have changed somehow. let me start from scratch and see where it goes...

**Comment by nicolas-chaulet** [comment]
yes we definitely had an issue, should be fixed when #441 is merged, thanks for reporting!

**Comment by wwwwwlllllllllll** [comment]
The problem comes beause the PointCloudFusion wthin pre_collate_trasform in s3disfused-fixed.yaml，the PointCloudFusion use the torch_geometric.Data.Batch.data_from_list function and create a list of fused point cloud with the attribute ptr. The size of attribute ptr is different, the size of a fused point cloud's ptr is 38, another is 34.
I fix this problem by remove the key ptr in the batch.py.
